django/core/management/commands/shell.py

```diff
@@ -111,7 +126,22 @@ def python(self, options):
         # Start the interactive interpreter.
         code.interact(local=imported_objects)
 
+    def get_namespace(self):
+        apps_models = apps.get_models()
+        imported_objects = {}
+        for model in reversed(apps_models):
```

Comment:
No `reversed()` so that earlier apps take precedence. Test precedence.

Reply:
Hi @adamchainz, I'm sorry to bother you with this question, but I have a doubt about testing the precedence for `get_namespace`. When you say earlier apps take precedence, do you mean that if we have a name collision, we have to import the models listed first in `INSTALLED_APPS`?

Reply:
Yes, in a collision the earlier model should be the one in the namespace. This follows the precedence order of other name resolutions in Django, such as for management commands.

As for second, replaced models, it would be nice to provide some way of still providing access, such as by having each app’s models module imported as `<app_label>_models`, so one could do e.g. `auth_models.User`. But let’s have a forum discussion about that first, since there are many different strategies we could copy from django-extensions.

Reply:
> No `reversed()` so that earlier apps take precedence. Test precedence.

Hi @adamchainz,

Thank you for your advice. I'll be raising a discussion in the next few days on what strategy to adopt for name collisions.

Apart from that, I've tested precedence with specific tests locally, and I've observed that if we loop over apps_models, we aren't actually giving precedence to the earlier apps because the latest will override the earlier ones. Instead, testing the precedence using reversed() seems to respect the precedence of the earlier apps, which will override the latest listed in the list.

Please let me know if I am saying something incorrect.

Reply:
You were correct, I maybe misread and thought the loop was doing a check like `if model.__name__ not in namespace`.